src/model/transaction/splitBlockInContentState.js

```diff
@@ -13,55 +13,125 @@
 
 'use strict';
 
+import type {BlockMap} from 'BlockMap';
 import type ContentState from 'ContentState';
 import type SelectionState from 'SelectionState';
 
-var Immutable = require('immutable');
+const ContentBlockNode = require('ContentBlockNode');
+const Immutable = require('immutable');
 
-var generateRandomKey = require('generateRandomKey');
-var invariant = require('invariant');
+const generateRandomKey = require('generateRandomKey');
+const invariant = require('invariant');
 
-const {Map} = Immutable;
+const {List, Map} = Immutable;
 
-function splitBlockInContentState(
+const transformBlock = (
```

Comment:
Worth noting that his is the exact same function used on the remove ranges PR (https://github.com/mitermayer/draft-js/blob/d151ad92d52c0548156a81a7fec399bb2da3d6ea/src/model/transaction/removeRangeFromContentState.js#L28) 

We can either rebase either PR's to which one is merged first or do a follow up PR to remove the duplication

Reply:
We should also during that process refactor moveBlockInContentState to also use `transformBlock`

Reply:
Doing a follow-up seems like a good idea - since these are scoped, well defined tasks we can also open issues for folks in the community to DRY up the code.